lightning/src/chain/channelmonitor.rs

```diff
@@ -3040,6 +3116,92 @@ impl<Signer: EcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 			}
 		}
 	}
+	
+	/// Same as [provide_payment_preimage], but does not immediately broadcasts transactions,
+	/// returning them back to the caller instead.
+	fn provide_payment_preimage_without_broadcasting<L: Deref>(
```

Comment:
@TheBlueMatt Not sure about this func. On one hand I don't want to change `provide_payment_preimage`, on the other -- there is clear duplicated logic. Do you think it would be better to modify `provide_payment_preimage` to support returning claims?